telegram/ext/_dispatcher.py

```diff
@@ -573,11 +571,53 @@ def add_handler(self, handler: Handler[UT, CCT], group: int = DEFAULT_GROUP) ->
 
         if group not in self.handlers:
             self.handlers[group] = []
-            self.groups.append(group)
-            self.groups = sorted(self.groups)
+            self.handlers = dict(sorted(self.handlers.items()))  # lower -> higher groups
 
         self.handlers[group].append(handler)
 
+    def add_handlers(
+        self,
+        handlers: Union[
+            Union[List[Handler], Tuple[Handler]], Dict[int, Union[List[Handler], Tuple[Handler]]]
+        ],
+        group: DVInput[int] = DefaultValue(0),
+    ) -> None:
+        """Registers multiple handlers at once. The order of the handlers in the passed
+        sequence(s) matters. See :meth:`add_handler` for details.
+
+        .. versionadded:: 14.0
+        .. seealso:: :meth:`add_handler`
+
+        Args:
+            handlers (List[:obj:`telegram.ext.Handler`] | \
+                Dict[int, List[:obj:`telegram.ext.Handler`]]): \
+                Specify a sequence of handlers *or* a dictionary where the keys are groups and
+                values are handlers.
+            group (:obj:`int`, optional): Specify which group the sequence of ``handlers``
+                should be added to. Defaults to ``0``.
+
+        """
+        if isinstance(handlers, dict) and not isinstance(group, DefaultValue):
+            raise ValueError('The `group` argument can only be used with a sequence of handlers.')
+
+        if isinstance(handlers, dict):
+            for handler_group, grp_handlers in handlers.items():
+                if not isinstance(grp_handlers, (list, tuple)):
+                    raise ValueError(f'Handlers for group {handler_group} must be a list or tuple')
+
+                for handler in grp_handlers:
+                    self.add_handler(handler, handler_group)
```

Comment:
we dont do a check before this if handler is actually an instance of Handler, which we do in the `add_handler` function. I would add this.

Reply:
but wouldn't that be redundant? It would raise an error in `add_handler` anyway.

Reply:
yeah, sure. no idea how I missed you literally call add_handler there. Nevermind this.